modules/plot/main/coffee/Axis.coffee

```diff
@@ -455,25 +466,30 @@ class Axis
         else
           markerY = height - yOffset - @xAxisSize
 
+      axisUpdateFunc = (scale) ->
+        @view('.hx-axis-line', 'line')
+          .update((s) ->
+            @attr('x1', s.rangeMin).attr('x2', s.rangeMax)
+            @attr('y1', axisY).attr('y2', axisY)
+          ).apply(scale)
+
+        @view('.hx-tick', 'g')
+          .update (tick, e, i) ->
+            @attr("transform", "translate(" + tick[1] + "," + markerY + ")")
+            @view('.hx-tick-line', 'line')
+              .update (t) -> @attr('y1', 0).attr('y2', tickSize)
+              .apply(this)
+            @view('.hx-tick-text-x', 'text')
+              .update (t) -> @text(if (i%self.x.nthTickVisible())==0 then self.x.formatter()(t) else '')
+              .apply(tick[0])
+          .apply(if self.x.showTicks() then scale.ticks(self.x.tickSpacing()) else [])
+
+        if self.x.doCollisionDetection()
+          doCollisionDetection @selectAll('.hx-tick-text-x').filter((x) -> x.text()).nodes
```

Comment:
Should probably check `self.x.showTicks()` and the new option value in the if statement.

In the case that showTicks is off currently, we'd effectively call `doCollsionDetection([])` after selecting nothing